File: src/main/java/pl/training/workshop/m7/s01_breakdependencies/step3/LegacyDatabase.java
package pl.training.workshop.m7.s01_breakdependencies.step3;

import java.util.List;

import pl.training.workshop.m7.s01_breakdependencies.PaidBooking;

/** Produkcyjna baza. Konstruktor otwiera połączenie - poza serwerownią kina to się nie uda. */
public final class LegacyDatabase implements BookingStore {
    public LegacyDatabase() {
        throw new IllegalStateException("brak polaczenia z jdbc:oracle:thin:@prod-db:1521/CINEMA");
    }

    @Override
    public List<PaidBooking> paidBookings() {
        return List.of();
    }

    @Override
    public void markReminded(String bookingId) {
        // UPDATE bookings SET reminded = 1 WHERE id = ?
    }
}

File: csharp/src/Training.Workshop/M7/S01BreakDependencies/Step3/LegacyDatabase.cs
namespace Training.Workshop.M7.S01BreakDependencies.Step3;

/// <summary>Produkcyjna baza. Konstruktor otwiera połączenie - poza serwerownią kina to się nie uda.</summary>
public sealed class LegacyDatabase : IBookingStore
{
    public LegacyDatabase()
    {
        throw new InvalidOperationException("brak polaczenia z jdbc:oracle:thin:@prod-db:1521/CINEMA");
    }

    public IReadOnlyList<PaidBooking> PaidBookings()
    {
        return [];
    }

    public void MarkReminded(string bookingId)
    {
        // UPDATE bookings SET reminded = 1 WHERE id = ?
    }
}
